actionpack/lib/action_dispatch/http/content_security_policy.rb

```diff
@@ -24,11 +24,32 @@ module ActionDispatch # :nodoc:
   #
   #       # Specify URI for violation reports
   #       policy.report_uri "/csp-violation-report-endpoint"
+  #       policy.report_to 'default', -> {
+  #         {
+  #           default: {
+  #             urls: ['/csp-violation-report-endpoint', 'https://example.com/csp-violation-report'],
+  #             max_age: 30.minutes,
+  #             include_subdomains: true
+  #           },
+  #           group_2: 'https://example.com/hpkp-reports'
+  #         }
+  #       }
   #     end
+
   class ContentSecurityPolicy
     class InvalidDirectiveError < StandardError
     end
 
+    class ReportingEndpointError < StandardError
+      def initialize(message, provided_data = nil)
+        if provided_data
+          super("#{message} Provided data: #{provided_data.inspect}")
```

Comment:
```suggestion
          super("#{message}, provided data: #{provided_data.inspect}")
```

Maybe? Since we are concatenating messages.